update_python_packages.py

```diff
@@ -0,0 +1,111 @@
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import subprocess
+import os
+import sys
+
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from zipfile import ZipFile
+
+
+PYTHON_VERSION = f"{sys.version_info.major}.{sys.version_info.minor}"
+
+
+def zip_recursively(zip_file, root_dir, folder_name):
+    for root, _, files in os.walk(root_dir / folder_name):
+        for file in files:
+            full_file_path = Path(os.path.join(root, file))
+            zip_file.write(full_file_path, full_file_path.relative_to(root_dir))
+
+
+def ensure_init_file(package_dir):
+    """
+    Ensures that every package contains an __init__.py file.
+    This is required for Python to recognize the directory as a package.
+    """
+    for root, _, files in os.walk(package_dir):
+        if "__init__.py" not in files:
+            init_file_path = Path(root) / "__init__.py"
+            with open(init_file_path, "w") as f:
+                f.write("")
+
+
+with TemporaryDirectory() as temp_dir:
+    temp_dir = Path(temp_dir)
+
+    # Make sure the requirements folder exists
+    if not os.path.exists(f"requirements/{PYTHON_VERSION}/requirements.txt"):
+        raise RuntimeError(f"Python {PYTHON_VERSION} requirements not found.")
+
+    # Pip install everything and capture everything that was installed.
+    print(f"Installing Python {PYTHON_VERSION} requirements...")
+    subprocess.run(
+        [
+            "python",
+            "-m",
+            "pip",
+            "install",
+            "-r",
+            f"requirements/{PYTHON_VERSION}/requirements.txt",
+            "--no-compile",
+            # The combination of --target and --upgrade forces pip to install
+            # all packages to the temporary directory, even if an already existing
+            # version is installed
+            "--target",
+            str(temp_dir),
+            "--upgrade",
+        ]
+    )
+    print("Writing out frozen requirements...")
+    subprocess.run(
+        ["python", "-m", "pip", "freeze", "--path", str(temp_dir)],
+        stdout=open(f"requirements/{PYTHON_VERSION}/frozen_requirements.txt", "w"),
+    )
+    # Quickly compute the number of requirements we have.
+    nb_dependencies = len(
+        [
+            _
+            for _ in open(
+                f"requirements/{PYTHON_VERSION}/frozen_requirements.txt", "rt"
+            )
+        ]
+    )
+
+    # Figure out if those packages were installed as single file packages or folders.
+    package_names = [
+        package_name
+        for package_name in os.listdir(temp_dir)
+        if all(
+            [
+                "info" not in package_name,
+                package_name != "bin",
+                ".pyd" not in package_name,
```

Comment:
Are we really using any of that logic?

Could you compare your script with this one we rewrote recently please? https://github.com/shotgunsoftware/tk-framework-desktopclient/blob/master/dev/update_python_packages.py